internal/testtypes/builders.go

```diff
@@ -0,0 +1,525 @@
+		if err := setNestedRef(obj, c.infrastructureMachineTemplate, "spec", "template", "spec", "machineTemplate", "infrastructureRef"); err != nil {
+			panic(err)
+		}
+	}
+	return obj
+}
+
+// InfrastructureClusterBuilder holds the variables and objects needed to build a generic object for cluster.spec.infrastructureRef.
+type InfrastructureClusterBuilder struct {
+	namespace string
+	name      string
+}
+
+// NewInfrastructureClusterBuilder returns and InfrastructureClusterBuilder with the given name and namespace.
+func NewInfrastructureClusterBuilder(namespace, name string) *InfrastructureClusterBuilder {
+	return &InfrastructureClusterBuilder{
+		namespace: namespace,
+		name:      name,
+	}
+}
+
+// Build returns an Unstructured object with the information passed to the InfrastructureClusterBuilder.
+func (i *InfrastructureClusterBuilder) Build() *unstructured.Unstructured {
+	obj := &unstructured.Unstructured{}
+	obj.SetAPIVersion(InfrastructureGroupVersion.String())
+	obj.SetKind(GenericInfrastructureClusterKind)
+	obj.SetNamespace(i.namespace)
+	obj.SetName(i.name)
+
+	if err := unstructured.SetNestedField(obj.UnstructuredContent(), true, "spec", "fakeSetting"); err != nil {
+		panic(err)
+	}
+	fmt.Println(obj)
+	return obj
+}
+
+// ControlPlaneBuilder holds the variables and objects needed to build a generic object for cluster.spec.controlPlaneRef.
+type ControlPlaneBuilder struct {
+	namespace                     string
+	name                          string
+	infrastructureMachineTemplate *unstructured.Unstructured
+}
+
+// NewControlPlaneBuilder returns a ControlPlaneBuilder with the given name and Namespace.
+func NewControlPlaneBuilder(namespace, name string) *ControlPlaneBuilder {
+	return &ControlPlaneBuilder{
+		namespace: namespace,
+		name:      name,
+	}
+}
+
+// WithInfrastructureMachineTemplate adds the given unstructured object to the ControlPlaneBuilder as its InfrastructureMachineTemplate.
+func (f *ControlPlaneBuilder) WithInfrastructureMachineTemplate(t *unstructured.Unstructured) *ControlPlaneBuilder {
+	f.infrastructureMachineTemplate = t
+	return f
+}
+
+// Build generates an Unstructured object from the information passed to the ControlPlaneBuilder.
+func (f *ControlPlaneBuilder) Build() *unstructured.Unstructured {
+	obj := &unstructured.Unstructured{}
+	obj.SetAPIVersion(ControlPlaneGroupVersion.String())
+	obj.SetKind(GenericControlPlaneKind)
+	obj.SetNamespace(f.namespace)
+	obj.SetName(f.name)
+
+	if err := unstructured.SetNestedField(obj.UnstructuredContent(), true, "spec", "fakeSetting"); err != nil {
+		panic(err)
+	}
+
+	if f.infrastructureMachineTemplate != nil {
+		if err := setNestedRef(obj, f.infrastructureMachineTemplate, "spec", "machineTemplate", "infrastructureRef"); err != nil {
+			panic(err)
+		}
+	}
+	return obj
+}
+
+// MachineDeploymentBuilder holds the variables and objects needed to build a generic object for a clusterv1.MachineDeployment.
+type MachineDeploymentBuilder struct {
```

Comment:
Might make sense to add the WithReplicas function here

Reply:
👍